CDCgov/prime-reportstream
# 14599: initial creation of submissions/shared gradle project

This PR does some initial bootstrapping on the submissions sub-project as well as establishing a multi-project build.

Test Steps:
1. From the root (`prime-reportstream`) run `./gradlew quickRun` and verify the app starts
2. Run `./gradlew testSmoke` and confirm all the tests pass
3. Run `./gradlew bootRun` and verify you can visit http://localhost:8880/health
4. Verify you can run the app via docker

## Changes
- Adds a new submissions project initialized from spring boot and a single health endpoint
- Sets up the repo to be a multi-project gradle build
- Consolidates some of the gradle conventions into `buildSrc`

## Checklist

### Testing

### Process

## Linked Issues
- #14599 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
